firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -53,12 +62,18 @@
   private TaskCompletionSource<Void> signInTaskCompletionSource = null;
   private CancellationTokenSource signInCancellationSource;
 
+  private TaskCompletionSource<AppDistributionRelease> checkForUpdateTaskCompletionSource = null;
+  private CancellationTokenSource checkForUpdateCancellationSource;
+  private Handler checkForUpdateHandler;
+  private Executor checkForUpdateExecutor;
+
   /** Constructor for FirebaseAppDistribution */
   public FirebaseAppDistribution(
       @NonNull FirebaseApp firebaseApp,
       @NonNull FirebaseInstallationsApi firebaseInstallationsApi) {
     this.firebaseApp = firebaseApp;
     this.firebaseInstallationsApi = firebaseInstallationsApi;
```

Comment:
we should get additional feedback on this from some android experts :)

Reply:
Marc or someone from the A-core team will have more insights on this because I have seen other SDKs do something a little different

Reply:
I'll reach out!